tests/std/tests/P0896R4_P1614R2_comparisons/test.cpp

```diff
@@ -294,17 +294,8 @@ STATIC_ASSERT(test_compare_three_way<compares_as<strong_ordering>, compares_as<s
 // Validate dynamic properties of compare_three_way, ranges::equal_to, ranges::not_equal_to, ranges::less,
 // ranges::less_equal, ranges::greater, ranges::greater_equal
 template <class T, class U, class R>
-constexpr void assert_three_way(T const& t, U const& u, R const result) { // TRANSITION, VSO-905257
-#if !defined(__clang__) && !defined(__EDG__)
-    // <=> expressions that resolve to builtin operators are incorrectly lvalues of const-qualified type on MSVC, so
-    // they are rejected by std::three_way_comparable and std::three_way_comparable_with.
-    if constexpr (!std::is_class_v<std::remove_cvref_t<T>> && !std::is_class_v<std::remove_cvref_t<U>>) {
-        return;
-    } else
-#endif // !defined(__clang__) && !defined(__EDG__)
-    {
-        assert(compare_three_way{}(t, u) == result);
-    }
+constexpr void assert_three_way(T const& t, U const& u, R const result) {

```

Comment:
Is it worth keeping this template?

Reply:
No, it is not.

Reply:
I suggest `#define assert_three_way(t, u, result) assert((compare_three_way{}(t, u) == result)`.